src/libstd/fs.rs

```diff
@@ -756,6 +756,18 @@ impl AsInner<fs_imp::DirEntry> for DirEntry {
 /// guarantee that the file is immediately deleted (e.g. depending on
 /// platform, other open file descriptors may prevent immediate removal).
 ///
+/// # Platform Notes
+///
+/// Depending on platform, this function will invoke the following syscalls:
+///
+/// ### Windows
+///
+/// `DeleteFileW`
```

Comment:
I’m not sure specifying (and therefore stabilising) implementation details is a good thing to do or the way we want to go in.